Makefile

```diff
@@ -1,38 +1,18 @@
-.PHONY: git-submodules test bins clean cover cover_ci help
-PROJECT_ROOT = github.com/uber/cadence
-
-export PATH := $(shell go env GOPATH)/bin:$(PATH)
+# get rid of default behaviors, they're just noise
+MAKEFLAGS += --no-builtin-rules
+.SUFFIXES:
 
-ifndef GOOS
-GOOS := $(shell go env GOOS)
-endif
-
-ifndef GOARCH
-GOARCH := $(shell go env GOARCH)
-endif
+.PHONY: git-submodules test bins clean cover cover_ci help
+default: help
 
-THRIFT_GENDIR=.gen
+PROJECT_ROOT = github.com/uber/cadence
 
-default: help
+GOOS ?= $(shell go env GOOS)
+GOARCH ?= $(shell go env GOARCH)
 
-# define the list of thrift files the service depends on
-# (if you have some)
-THRIFTRW_SRCS = \
-  idls/thrift/cadence.thrift \
-  idls/thrift/health.thrift \
-  idls/thrift/history.thrift \
-  idls/thrift/matching.thrift \
-  idls/thrift/replicator.thrift \
-  idls/thrift/indexer.thrift \
-  idls/thrift/shared.thrift \
-  idls/thrift/admin.thrift \
-  idls/thrift/sqlblobs.thrift \
-  idls/thrift/checksum.thrift \
-
-PROGS = cadence
 TEST_TIMEOUT = 20m
 TEST_ARG ?= -race -v -timeout $(TEST_TIMEOUT)
-BUILD := ./build
+BUILD := .build
```

Comment:
You will also need to update https://github.com/uber/cadence/blob/master/.buildkite/pipeline.yml
to pick code coverage report from the new location.